projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-update-post-v1-2-endpoint.php

```diff
@@ -265,11 +265,12 @@ public function write_post( $path, $blog_id, $post_id ) {
 				$input['status'] = 'publish';
 			}
 
-			$post       = get_post( $post_id );
-			$_post_type = ( ! empty( $input['type'] ) ) ? $input['type'] : $post->post_type;
-			$post_type  = get_post_type_object( $_post_type );
+			$post = get_post( $post_id );
 			if ( ! $post || is_wp_error( $post ) ) {
 				return new WP_Error( 'unknown_post', 'Unknown post', 404 );
+			} else {
+				$_post_type = ( ! empty( $input['type'] ) ) ? $input['type'] : $post->post_type;
+				$post_type  = get_post_type_object( $_post_type );
 			}
```

Comment:
Could make it slightly more straightforward like this, since the `if` branch returns anyway.
```suggestion
			}

			$_post_type = ( ! empty( $input['type'] ) ) ? $input['type'] : $post->post_type;
			$post_type  = get_post_type_object( $_post_type );
```

Reply:
I was thinking about this as well so let's do it this way. Thanks for the suggestion!